google/cloud/spanner_v1/client.py

```diff
@@ -173,19 +174,28 @@ def __init__(
         client_options=None,
         query_options=None,
     ):
+        self._emulator_host = _get_spanner_emulator_host()
+
+        if client_options and type(client_options) == dict:
+            self._client_options = google.api_core.client_options.from_dict(
+                client_options
+            )
+        else:
+            self._client_options = client_options
+
+        if isinstance(credentials, AnonymousCredentials):
```

Comment:
for other libraries, we may have done the opposite.
If the emulator_env_variable is set, we hard-code anonymous/null credentials for the emulator.

Reply:
I have updated it to set AnonymousCredentials when the env var is set.